### PR TITLE
[WIP] Rewriting processor module with Kotlin

### PR DESCRIPTION
**Pull request by hotchemi**
## ToDo

- [x] Execute gradle uploadBintry command with dryRun

### CODE REVIEW

**Comment by hotchemi** [comment]
@aurae Sorry, what is current status?

**Comment by aurae** [comment]
As far as I (and the unit test suite) know, the switch to Kotlin is successful. That includes optional support for native Fragments based on the user's dependency configuration!

**Comment by hotchemi** [comment]
Oh boy it is amazing...it's quite difficult where to merge it...

**Comment by aurae** [comment]
Whenever you feel like all milestones for "version 1" have been reached and we are ready to move on to the _2.0.0_ architecture, we'll push the existing `master` into a separate branch (`version-1`, or `legacy`, whatever naming fits best), clean up the current master and merge from the Kotlin branch.

**Comment by hotchemi** [comment]
First of all, I wanna pass the travis ci build.

**Comment by aurae** [comment]
Sounds good, I understand. Let's do that first then! I will get on it asap

**Comment by aurae** [comment]
It took me a few tries of tweaking Travis CI's configuration, but now the merge conflicts are resolved and the CI tests pass for this new version. I also created the "version1" branch which holds the current stable version of the library (`1.2.1`).

**Comment by hotchemi** [comment]
Oh, good!
Just a minute, I'd like to check it carefully, because I don't wanna causes regression.

**Comment by aurae** [comment]
cont'd from #36:
> And when it comes to the Kotlin, I've been thinking which we should go, Java8 or Kotlin...

Seeing as the implementation in Kotlin is already done and working, I don't see why we would do it all over again in Java 8. In my experience writing the processor, Kotlin fits very well into an annotation processing environment.

**Comment by hotchemi** [comment]
Of course Kotlin is great, but I wanna choose really fit one.

## Merit

- Kotlin simplifies the logic.

## Demerit

- Think about contribution, Java is still good one because almost android developers understand Java.
- Kotlin uses more memory than usual Java.
- When it comes to follow update, Kotlin is still doing destructive evolution so it may be hard to follow.

**Comment by aurae** [comment]
I have updated the code to conform to Kotlin 1.0.0 syntax. I have also implemented app-wide unique request codes in response to our discussion in Tokyo. :)

**Comment by hotchemi** [comment]
I can't merge it via PR so I close it and merge the branch manually.
